chore: mac os 13 hosted runner image is closing down

ianna opened the pull request:
The macOS 13 runner image will be retired by December 4th, 2025. To raise awareness of the upcoming removal, jobs using macOS 13 will temporarily fail during the scheduled brownout time periods defined below:

    November 4, 14:00 UTC - November 5, 00:00 UTC
    November 11, 14:00 UTC - November 12, 00:00 UTC
    November 18, 14:00 UTC - November 19, 00:00 UTC
    November 25, 14:00 UTC - November 26, 00:00 UTC

This deprecation includes the following labels:

    macos-13
    macos-13-large
    macos-13-xlarge

ianna committed:
migrate to arm64 label

ianna committed:
Update header-only-test.yml

ianna committed:
Update packaging-test.yml

ianna committed:
Update build-wheels.yml

ariostas commented on .github/workflows/test.yml:
```diff
@@ -37,7 +37,7 @@ jobs:
           - windows-latest
           - ubuntu-latest
-          - macos-13
+          - macos-latest
```
```suggestion
          - macos-14
```
I think this might fix the CI errors

ianna committed:
Update .github/workflows/test.yml

Co-authored-by: Andres Rios Tascon <ariostas@gmail.com>

ianna committed:
Merge branch 'main' into ianna/macOS_13_hosted_runner_image_is_closing_down

ianna committed:
Merge branch 'main' into ianna/macOS_13_hosted_runner_image_is_closing_down

ianna committed:
Merge branch 'main' into ianna/macOS_13_hosted_runner_image_is_closing_down

ianna committed:
Update test.yml to include more Python versions

Added support for multiple Python versions on macOS 14 with arm64 architecture.

ianna committed:
Exclude macOS 14 x64 from workflow tests

ianna committed:
Add dependencies-kind field to macOS job

ianna committed:
Clean up macOS Python version configurations

Removed multiple macOS configurations for Python versions 3.9 to 3.14.

ianna committed:
Merge branch 'main' into ianna/macOS_13_hosted_runner_image_is_closing_down
